src/ser.rs

```diff
@@ -425,7 +424,7 @@ impl<'a> Serializer<'a> {
 
     fn display<T: fmt::Display>(&mut self, t: T, type_: &'static str) -> Result<(), Error> {
         self.emit_key(type_)?;
-        drop(write!(self.dst, "{}", t));
+        write!(self.dst, "{}", t);
```

Comment:
I think this should probably be `?` or remain as `drop` perhaps?

Reply:
(same for all other instances of this)

Reply:
I think the `?` approach is more idiomatic, errors are now being mapped to `ser::Error` via `ser::Error::custom`

If all is well, I'm going to open this as an actual PR